CRM/Dedupe/BAO/DedupeRuleGroup.php

```diff
@@ -14,12 +14,19 @@
  * @package CRM
  * @copyright CiviCRM LLC https://civicrm.org/licensing
  */
+use Civi\Core\HookInterface;
 
 /**
  * The CiviCRM duplicate discovery engine is based on an
  * algorithm designed by David Strauss <david@fourkitchens.com>.
  */
-class CRM_Dedupe_BAO_DedupeRuleGroup extends CRM_Dedupe_DAO_DedupeRuleGroup {
+class CRM_Dedupe_BAO_DedupeRuleGroup extends CRM_Dedupe_DAO_DedupeRuleGroup implements \Symfony\Component\EventDispatcher\EventSubscriberInterface {
```

Comment:
@eileenmcnaughton this should be the \Civi\Core\HookInterface not Event Subscriber Interface

Reply:
@seamuslee001 I didn't really figure out which is which based on the docs - but this one seems to allow me to specify the weight

Reply:
As we're using weight / getSubscribedEvents() it does need to be EventSubscriberInterface